tests/integration/test_enrollment/conftest.py

```diff
@@ -0,0 +1,333 @@
+    """
+    if len(received) == 0:
+        return b""
+
+    global LAST_MESSAGE
+    LAST_MESSAGE = received.decode()
+    response = CURRENT_TEST_CASE['message']['response'].format(**DEFAULT_VALUES).encode()
+    return response
+
+
+def get_last_message():
+    """Returns the last received message in the listener socket. Waits 20 seconds of timeout.
+    Returns:
+        LAST_MESSAGE (str): Decoded data received in the socket.
+        None if the response never arrived.
+    """
+    global LAST_MESSAGE
+    import time
+    timeout = time.time() + 20  # 20 seconds timeout
+    while not LAST_MESSAGE and time.time() <= timeout:
+        pass
+    return LAST_MESSAGE
+
+
+def clear_last_message():
+    """Clears the last received message in the listener socket to wait for a new message."""
+    global LAST_MESSAGE
+    LAST_MESSAGE = None
+
+
+socket_listener = ManInTheMiddle(address=(DEFAULT_VALUES['manager_address'], DEFAULT_VALUES['port']), family='AF_INET',
+                                 connection_protocol='SSL', func=receiver_callback)
+
+
+@pytest.fixture(scope="module")
+def create_certificates():
+    cert_controller = CertificateController()
+    cert_controller.get_root_ca_cert().sign(cert_controller.get_root_key(), cert_controller.digest)
+    cert_controller.store_private_key(cert_controller.get_root_key(), AGENT_KEY_PATH)
+    cert_controller.store_ca_certificate(cert_controller.get_root_ca_cert(), AGENT_CERT_PATH)
+
+
+@pytest.fixture(scope="function")
+def configure_socket_listener():
+    """Configures the socket listener to start listening on the socket."""
+    socket_listener.start()
+    socket_listener.listener.set_ssl_configuration(connection_protocol=ssl.PROTOCOL_TLSv1_2,
+                                                   certificate=AGENT_CERT_PATH,
+                                                   keyfile=AGENT_KEY_PATH,
+                                                   options=None,
+                                                   cert_reqs=ssl.CERT_OPTIONAL)
+
+    while not socket_listener.queue.empty():
+        socket_listener.queue.get_nowait()
+    socket_listener.event.clear()
+
+    yield
+    socket_listener.shutdown()
+
+
+# Wazuh conf
+
+def get_temp_yaml(param):
+    """Creates a temporal config file."""
+    temp = os.path.join(test_data_path, 'temp.yaml')
+    with open(configurations_path, 'r') as conf_file:
+        enroll_conf = {'enrollment': {'elements': []}}
+        for elem in param:
+            if elem == 'password':
+                continue
+            enroll_conf['enrollment']['elements'].append({elem: {'value': param[elem]}})
+        print(enroll_conf)
+        temp_conf_file = yaml.safe_load(conf_file)
+        temp_conf_file[0]['sections'][0]['elements'].append(enroll_conf)
+    with open(temp, 'w') as temp_file:
+        yaml.safe_dump(temp_conf_file, temp_file)
+    return temp
+
+
+def override_wazuh_conf(configuration, test):
```

Comment:
`get_configuration` and `configure_environment` already do this.

Reply:
**configure_environment** is designed to implement the configuration from **wazuh_enrollment_conf.yaml** and is one configuration per test, not per test_case.
On the other hand, **override_wazuh_conf** is implemented to override the Wazuh configuration for each test_case from wazuh_enrollment_tests.yaml. 
Is there an example of the approach that you suggest?